dbms/src/Encryption/KeyspacesKeyManager.cpp

```diff
@@ -40,11 +47,11 @@ FileEncryptionInfo KeyspacesKeyManager<ProxyHelper>::getInfo(const EncryptionPat
         };
     }
 
-    RUNTIME_CHECK(ps_write != nullptr);
+    RUNTIME_CHECK(!ps_write.expired());
```

Comment:
Should use weak_ptr as follows. Or there is still a chance that the instance is released between you calling `ps_write.expired()` and `ps_write.lock()`.
```
auto ps_write_locked = ps_write.lock();
RUNTIME_CHECK(!ps_write_locked);
...
ps_write_locked->read(...);
```

Reply:
done